### PR TITLE
Add unknown event instead of error on audit log read

### PR DESCRIPTION
**Pull request by xacrimon**
This PR adds an unknown audit event. This is used to contain data about events with unknown types when they are found when reading from the backend among other places. This prevents ugly hard-error behaviour in the case of a bug and maintains accessibility to the audit log. In addition, it makes debugging these cases easier since we now emit an error log where it happens, when and in addition, the unknown events contains the JSON-serialized data of the event, allowing for quickly finding the erroneous emit point by filtering on the other data.

### CODE REVIEW

File: api/types/events/events.proto [hunk 1]
```diff
@@ -1782,6 +1782,16 @@ message RenewableCertificateGenerationMismatch {
         [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
 }
 
+// Unknown is a fallback event used when we don't recognize an event from the backend.
+message Unknown {
+    // Metadata is a common event metadata.
+    Metadata Metadata = 1
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
```
Comment (zmb3) [inline api/types/events/events.proto:1790]: Should we consider adding the code/name of the event in strongly-typed fields?

We could try to populate these on a best-effort basis by attempting to unmarshal the `event` and `code` fields from the JSON data.

The advantage of this is the description in the audit log could then read something like:

`An unknown "foo.bar.baz" (T0123) event`

Comment (xacrimon) [reply api/types/events/events.proto:1790]: @zmb3 Great idea, added it.

File: api/types/events/oneof.go [hunk 2]
```diff
@@ -353,8 +355,28 @@ func ToOneOf(in AuditEvent) (*OneOf, error) {
...
+	case *Unknown:
+		out.Event = &OneOf_Unknown{
+			Unknown: e,
+		}
 	default:
-		return nil, trace.BadParameter("event type %T is not supported", in)
+		log.Errorf("Attempted to convert dynamic event of unknown type \"%v\" into protobuf event.", in.GetType())
+		unknown := &Unknown{}
+		unknown.Type = "unknown"
+		unknown.Code = "TCC00E"
```
Comment (xacrimon) [inline api/types/events/oneof.go:366]: note to reviewers: this isn't pretty but is unavoidable in the absence of significant package refactoring due to cyclic dependencies

Comment (zmb3) [reply api/types/events/oneof.go:366]: Can you define these in api/types/events then just alias them in lib/events?

```
UnknownCode = events.UnknownCode
```

Comment (xacrimon) [reply api/types/events/oneof.go]: Sure, that at least saves us constant duplication

**Review by zmb3 — APPROVED** [review]
Code looks good to me. Would be good to test by temporarily running a dev build that emits some new event to the logs, and then "reverting" to this branch and seeing what happens.

**Review by timothyb89 — APPROVED** [review]

**Pull request merged**